[RAC] [Metrics UI] Include group name in the reason message

mgiota opened the pull request:
Fixes https://github.com/elastic/kibana/issues/109610
Fixes https://github.com/elastic/kibana/issues/115469

**Before:**

<img width="1510" alt="Screenshot 2021-10-15 at 14 15 50" src="https://user-images.githubusercontent.com/2852703/137485173-5034307d-243c-41dd-9c47-5d447b5f5230.png">

**After:**
<img width="1552" alt="Screenshot 2021-10-15 at 14 11 37" src="https://user-images.githubusercontent.com/2852703/137484839-d5ab50b0-abf9-408f-b2a4-2a625a886f92.png">

elasticmachine commented:
Pinging @elastic/logs-metrics-ui (Team:logs-metrics-ui)

claudiopro commented:
There's also a type check failure in a type defined in this module. Looks like we could import and reuse a common type definition?

https://github.com/elastic/kibana/blob/8cd19cedbaea37c8493df1a9b4db9a1d16d1ccf0/x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts#L250-L260

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161958/)
* Commit: 065de0af4ae60994009f1b6653be02320d70ec97
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161958/execution/node/415/log)

### History
* :broken_heart: [Build #161902](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161902/) failed 88fc6eee9c4f8e8ebfd0e797794ef557412bae14
* :broken_heart: [Build #161391](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161391/) failed 8cd19cedbaea37c8493df1a9b4db9a1d16d1ccf0
* :broken_heart: [Build #161376](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161376/) failed 2d3c2c92493b0ad782d4310f63ec00b8b6b7913d
* :broken_heart: [Build #161359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161359/) failed b91716db36983f554f55b7b0f1326a0b813f2e5f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @mgiota

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161359/","number":"161359","commit":"b91716db36983f554f55b7b0f1326a0b813f2e5f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161376/","number":"161376","commit":"2d3c2c92493b0ad782d4310f63ec00b8b6b7913d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161391/","number":"161391","commit":"8cd19cedbaea37c8493df1a9b4db9a1d16d1ccf0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161902/","number":"161902","commit":"88fc6eee9c4f8e8ebfd0e797794ef557412bae14"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161958/","number":"161958","commit":"065de0af4ae60994009f1b6653be02320d70ec97"}]}
PIPELINE-->

mgiota commented:
@claudiopro Yep makes sense what you suggest . We discussed in the past about a similar topic regarding how we could add stricter rule types in metrics and reuse stuff. This could actually be something you could work if you wish. If we can get this PR merged without this change that would be great, since we have code freeze tomorrow.

Apart from inventory rule type I did a few more testing with metrics rule type as well and the reason message didn't look that good (check first alert)
<img width="2080" alt="Screenshot 2021-10-18 at 12 44 23" src="https://user-images.githubusercontent.com/2852703/137720868-c12e02e8-9eaa-4697-a7c5-ae3e4ccc6805.png">

So I changed the reason format and appended the group name at the end, which is similar to how apm does it:

<img width="1941" alt="Screenshot 2021-10-18 at 12 48 12" src="https://user-images.githubusercontent.com/2852703/137721165-6b3a5471-1741-4cc9-852c-7afca8b213df.png">

@weltenwort I had to do some changes in translation files and specifically I added `for {group}` in this chinese file. https://github.com/mgiota/kibana/commit/065de0af4ae60994009f1b6653be02320d70ec97#diff-1fad1ffc662e15d9b52603af09bd4b9bff0b220304f4256e45a5ad92882ff74cR13394 . How do I ask for a translation for the `for` word in chinese?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162112/)
* Commit: 6949af2f70aff46b088bab5c942497ad46081d90
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162112/execution/node/428/log)

### History
* :broken_heart: [Build #161958](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161958/) failed 065de0af4ae60994009f1b6653be02320d70ec97
* :broken_heart: [Build #162015](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162015/) failed 3556efe5f4c8f71fdaf6e662eab460c433cd7c89
* :broken_heart: [Build #161902](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161902/) failed 88fc6eee9c4f8e8ebfd0e797794ef557412bae14
* :broken_heart: [Build #161391](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161391/) failed 8cd19cedbaea37c8493df1a9b4db9a1d16d1ccf0
* :broken_heart: [Build #161376](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161376/) failed 2d3c2c92493b0ad782d4310f63ec00b8b6b7913d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @mgiota

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161376/","number":"161376","commit":"2d3c2c92493b0ad782d4310f63ec00b8b6b7913d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161391/","number":"161391","commit":"8cd19cedbaea37c8493df1a9b4db9a1d16d1ccf0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161902/","number":"161902","commit":"88fc6eee9c4f8e8ebfd0e797794ef557412bae14"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162015/","number":"162015","commit":"3556efe5f4c8f71fdaf6e662eab460c433cd7c89"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161958/","number":"161958","commit":"065de0af4ae60994009f1b6653be02320d70ec97"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162112/","number":"162112","commit":"6949af2f70aff46b088bab5c942497ad46081d90"}]}
PIPELINE-->

weltenwort commented:
@mgiota the only manual change to the translation `json` files we should make is the removal of entries to which no reference exists anymore. And that change should be made solely using the `node scripts/i18n_check.js --fix` script.

Aside from that we're only changing the `defaultMessage` properties in the actual code. All other translation work will be handled by dedicated specialists.

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162357/)
* Commit: 7f002c356b2cdb8186051856343530ab544a67a9
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-115171/7f002c356b2cdb8186051856343530ab544a67a9)

* Flaky suites:
  * xpack-kibana-ciGroup10

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162357/testReport)

<details><summary>Kibana Pipeline / general / Chrome X-Pack UI Functional Tests.x-pack/test/functional/apps/uptime/ping_redirects·ts.Uptime app with generated data but no data reset Ping redirects loads and goes to details page</summary>

[Link to Jenkins](https://kibana-ci.elastic.co//job/elastic+kibana+pipeline-pull-request/162357/testReport/junit/Chrome%20X-Pack%20UI%20Functional%20Tests/x-pack_test_functional_apps_uptime_ping_redirects·ts/Kibana_Pipeline___general___Uptime_app_with_generated_data_but_no_data_reset_Ping_redirects_loads_and_goes_to_details_page)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 6 times on tracked branches: https://github.com/elastic/kibana/issues/84992

[00:00:00]     │
[00:42:05]       └-: Uptime app
[00:42:05]         └-> "before all" hook in "Uptime app"
[00:47:16]         └-: with generated data but no data reset
[00:47:16]           └-> "before all" hook in "with generated data but no data reset"
[00:47:16]           └-: Ping redirects
[00:47:16]             └-> "before all" hook for "loads and goes to details page"
[00:47:16]             └-> "before all" hook for "loads and goes to details page"
[00:47:16]               │ info [x-pack/test/functional/es_archives/uptime/blank] Loading "mappings.json"
[00:47:16]               │ info [x-pack/test/functional/es_archives/uptime/blank] Loading "data.json"
[00:47:16]               │ info [o.e.c.m.MetadataCreateIndexService] [node-01] [heartbeat-8-generated-test] creating index, cause [api], templates [], shards [1]/[1]
[00:47:16]               │ info [x-pack/test/functional/es_archives/uptime/blank] Created index "heartbeat-8-generated-test"
[00:47:16]               │ debg [x-pack/test/functional/es_archives/uptime/blank] "heartbeat-8-generated-test" settings undefined
[00:47:16]             └-> loads and goes to details page
[00:47:16]               └-> "before each" hook: global before each for "loads and goes to details page"
[00:47:16]               └-> "before each" hook: delete settings for "loads and goes to details page"
[00:47:16]                 │ debg Deleting saved object {
[00:47:16]                 │        type: 'uptime-dynamic-settings',
[00:47:16]                 │        id: 'uptime-dynamic-settings-singleton'
[00:47:16]                 │      }/%s
[00:47:17]               └-> "before each" hook for "loads and goes to details page"
[00:47:18]               │ debg TestSubjects.exists(uptimeSettingsToOverviewLink)
[00:47:18]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="uptimeSettingsToOverviewLink"]') with timeout=0
[00:47:18]               │ debg --- retry.tryForTime error: [data-test-subj="uptimeSettingsToOverviewLink"] is not displayed
[00:47:19]               │ debg navigating to uptime url: http://localhost:61231/app/uptime
[00:47:19]               │ debg navigate to: http://localhost:61231/app/uptime
[00:47:19]               │ debg browser[INFO] http://localhost:61231/app/uptime?_t=1634600334760 281 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:47:19]               │
[00:47:19]               │ debg browser[INFO] http://localhost:61231/bootstrap.js 41:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:47:19]               │ debg ... sleep(700) start
[00:47:19]               │ debg ... sleep(700) end
[00:47:19]               │ debg returned from get, calling refresh
[00:47:20]               │ debg browser[INFO] http://localhost:61231/app/uptime?_t=1634600334760 281 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:47:20]               │
[00:47:20]               │ debg browser[INFO] http://localhost:61231/bootstrap.js 41:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:47:21]               │ debg currentUrl = http://localhost:61231/app/uptime
[00:47:21]               │          appUrl = http://localhost:61231/app/uptime
[00:47:21]               │ debg TestSubjects.find(kibanaChrome)
[00:47:21]               │ debg Find.findByCssSelector('[data-test-subj="kibanaChrome"]') with timeout=60000
[00:47:21]               │ debg ... sleep(501) start
[00:47:21]               │ debg ... sleep(501) end
[00:47:21]               │ debg in navigateTo url = http://localhost:61231/app/uptime
[00:47:21]               │ debg isGlobalLoadingIndicatorVisible
[00:47:21]               │ debg TestSubjects.exists(globalLoadingIndicator)
[00:47:21]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="globalLoadingIndicator"]') with timeout=1500
[00:47:23]               │ debg --- retry.tryForTime error: [data-test-subj="globalLoadingIndicator"] is not displayed
[00:47:23]               │ debg TestSubjects.exists(globalLoadingIndicator-hidden)
[00:47:23]               │ debg Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000
[00:47:23]               │ debg TestSubjects.exists(uptimeOverviewPage)
[00:47:23]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="uptimeOverviewPage"]') with timeout=2000
[00:47:23]               │ debg isGlobalLoadingIndicatorVisible
[00:47:23]               │ debg TestSubjects.exists(globalLoadingIndicator)
[00:47:23]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="globalLoadingIndicator"]') with timeout=1500
[00:47:25]               │ debg --- retry.tryForTime error: [data-test-subj="globalLoadingIndicator"] is not displayed
[00:47:25]               │ debg TestSubjects.exists(globalLoadingIndicator-hidden)
[00:47:25]               │ debg Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000
[00:47:25]               │ debg TestSubjects.exists(superDatePickerToggleQuickMenuButton)
[00:47:25]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="superDatePickerToggleQuickMenuButton"]') with timeout=20000
[00:47:26]               │ debg TestSubjects.exists(superDatePickerShowDatesButton)
[00:47:26]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="superDatePickerShowDatesButton"]') with timeout=2500
[00:47:26]               │ debg TestSubjects.click(superDatePickerShowDatesButton)
[00:47:26]               │ debg Find.clickByCssSelector('[data-test-subj="superDatePickerShowDatesButton"]') with timeout=10000
[00:47:26]               │ debg Find.findByCssSelector('[data-test-subj="superDatePickerShowDatesButton"]') with timeout=10000
[00:47:26]               │ debg TestSubjects.exists(superDatePickerstartDatePopoverButton)
[00:47:26]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="superDatePickerstartDatePopoverButton"]') with timeout=2500
[00:47:26]               │ debg TestSubjects.getVisibleText(superDatePickerstartDatePopoverButton)
[00:47:26]               │ debg TestSubjects.find(superDatePickerstartDatePopoverButton)
[00:47:26]               │ debg Find.findByCssSelector('[data-test-subj="superDatePickerstartDatePopoverButton"]') with timeout=10000
[00:47:26]               │ debg TestSubjects.getVisibleText(superDatePickerendDatePopoverButton)
[00:47:26]               │ debg TestSubjects.find(superDatePickerendDatePopoverButton)
[00:47:26]               │ debg Find.findByCssSelector('[data-test-subj="superDatePickerendDatePopoverButton"]') with timeout=10000
[00:47:26]               │ debg TestSubjects.click(superDatePickerApplyTimeButton)
[00:47:26]               │ debg Find.clickByCssSelector('[data-test-subj="superDatePickerApplyTimeButton"]') with timeout=10000
[00:47:26]               │ debg Find.findByCssSelector('[data-test-subj="superDatePickerApplyTimeButton"]') with timeout=10000
[00:47:26]               │ debg TestSubjects.exists(uptimeMonitorPage)
[00:47:26]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="uptimeMonitorPage"]') with timeout=0
[00:47:26]               │ debg --- retry.tryForTime error: [data-test-subj="uptimeMonitorPage"] is not displayed
[00:47:26]               │ debg TestSubjects.click(monitor-page-link-redirect-testing-id)
[00:47:26]               │ debg Find.clickByCssSelector('[data-test-subj="monitor-page-link-redirect-testing-id"]') with timeout=10000
[00:47:26]               │ debg Find.findByCssSelector('[data-test-subj="monitor-page-link-redirect-testing-id"]') with timeout=10000
[00:47:27]               │ debg TestSubjects.exists(uptimeMonitorPage)
[00:47:27]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="uptimeMonitorPage"]') with timeout=30000
[00:47:29]               │ debg --- retry.tryForTime error: [data-test-subj="uptimeMonitorPage"] is not displayed
[00:47:32]               │ debg --- retry.tryForTime failed again with the same message...
[00:47:35]               │ debg --- retry.tryForTime failed again with the same message...
[00:47:38]               │ debg --- retry.tryForTime failed again with the same message...
[00:47:41]               │ debg --- retry.tryForTime failed again with the same message...
[00:47:44]               │ debg --- retry.tryForTime failed again with the same message...
[00:47:47]               │ debg --- retry.tryForTime failed again with the same message...
[00:47:50]               │ debg --- retry.tryForTime failed again with the same message...
[00:47:53]               │ debg --- retry.tryForTime failed again with the same message...
[00:47:56]               │ debg --- retry.tryForTime failed again with the same message...
[00:47:57]               │ info Taking screenshot "/dev/shm/workspace/parallel/23/kibana/x-pack/test/functional/screenshots/failure/Uptime app with generated data but no data reset Ping redirects loads and goes to details page.png"
[00:47:57]               │ info Current URL is: http://localhost:61231/app/uptime
[00:47:57]               │ info Saving page source to: /dev/shm/workspace/parallel/23/kibana/x-pack/test/functional/failure_debug/html/Uptime app with generated data but no data reset Ping redirects loads and goes to details page.html
[00:47:57]               └- ✖ fail: Uptime app with generated data but no data reset Ping redirects loads and goes to details page
[00:47:57]               │      Error: expected testSubject(uptimeMonitorPage) to exist
[00:47:57]               │       at TestSubjects.existOrFail (/dev/shm/workspace/parallel/23/kibana/test/functional/services/common/test_subjects.ts:45:13)
[00:47:57]               │       at Object.goToMonitor (test/functional/services/uptime/navigation.ts:65:9)
[00:47:57]               │       at UptimePageObject.loadDataAndGoToMonitorPage (test/functional/page_objects/uptime_page.ts:52:5)
[00:47:57]               │       at Context.<anonymous> (test/functional/apps/uptime/ping_redirects.ts:63:7)
[00:47:57]               │       at Object.apply (/dev/shm/workspace/parallel/23/kibana/node_modules/@kbn/test/target_node/functional_test_runner/lib/mocha/wrap_function.js:87:16)
[00:47:57]               │ 
[00:47:57]               │ 

```

#### Stack Trace
```
Error: expected testSubject(uptimeMonitorPage) to exist
    at TestSubjects.existOrFail (/dev/shm/workspace/parallel/23/kibana/test/functional/services/common/test_subjects.ts:45:13)
    at Object.goToMonitor (test/functional/services/uptime/navigation.ts:65:9)
    at UptimePageObject.loadDataAndGoToMonitorPage (test/functional/page_objects/uptime_page.ts:52:5)
    at Context.<anonymous> (test/functional/apps/uptime/ping_redirects.ts:63:7)
    at Object.apply (/dev/shm/workspace/parallel/23/kibana/node_modules/@kbn/test/target_node/functional_test_runner/lib/mocha/wrap_function.js:87:16)
```
</details>

---

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :yellow_heart: [Build #162269](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162269/) was flaky 8989f6bfdc4bc5c88be7f9c827b17d3607f1c031
* :green_heart: [Build #162134](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162134/) succeeded ee85ac3a5faccd104ef2e32b8be2d497667e420e
* :broken_heart: [Build #162112](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162112/) failed 6949af2f70aff46b088bab5c942497ad46081d90
* :yellow_heart: [Build #162029](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162029/) was flaky aaee1c3e32d0abb28fbe2d8dced8de6f718131dd
* :broken_heart: [Build #161958](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161958/) failed 065de0af4ae60994009f1b6653be02320d70ec97

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @mgiota

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161958/","number":"161958","commit":"065de0af4ae60994009f1b6653be02320d70ec97"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162029/","number":"162029","commit":"aaee1c3e32d0abb28fbe2d8dced8de6f718131dd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162112/","number":"162112","commit":"6949af2f70aff46b088bab5c942497ad46081d90"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162134/","number":"162134","commit":"ee85ac3a5faccd104ef2e32b8be2d497667e420e"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162269/","number":"162269","commit":"8989f6bfdc4bc5c88be7f9c827b17d3607f1c031"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/162357/","number":"162357","commit":"7f002c356b2cdb8186051856343530ab544a67a9"}]}
PIPELINE-->
